app/frontend/containers/CallToolView/CallToolView.js

```diff
@@ -147,18 +148,14 @@ class CallToolView extends Component {
     return _.sample(candidates);
   }
 
-  selectedTargetIndex() {
-    return _.findIndex(this.props.targets, this.state.selectedTarget);
-  }
-
   submit(event: any) {
     event.preventDefault();
     if(!this.validateForm()) return;
     this.setState({ errors: {}, loading: true });
     ChampaignAPI.calls.create({
       pageId: this.props.pageId,
       memberPhoneNumber: this.state.form.memberPhoneCountryCode + this.state.form.memberPhoneNumber,
-      targetIndex: this.selectedTargetIndex()
+      targetId: this.state.selectedTarget.id
```

Comment:
You're right. Perhaps the first alternative (return right after setState and before the call), but also setting the state back to `loading: false` in the return block?